aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -1043,23 +1043,38 @@ Tensor reduce_sparse_csr_dim0_cpu_template(const Tensor& sparse, ReductionOp rop
   new_crow_indices[1] = nnz;
 
   Tensor new_values = at::empty({nnz}, values.options());
-  new_values.fill_(rop.identity());
+  using opmath_t = at::opmath_type<scalar_t>;
+  constexpr bool need_acc = !std::is_same<scalar_t, opmath_t>::value;
+  Tensor new_values_acc;
+  if constexpr (need_acc) {
+    auto acc_dtype = values.scalar_type() == kComplexHalf
+        ? ScalarType::ComplexFloat
+        : ScalarType::Float;
+    new_values_acc = at::empty({nnz}, values.options().dtype(acc_dtype));
+  } else {
+    new_values_acc = new_values;
+  }
+  new_values_acc.fill_(rop.identity());
 
-  AT_DISPATCH_INDEX_TYPES(col_indices.scalar_type(), "reduce_sparse_csr_dim0_cpu_indices",
+  AT_DISPATCH_INDEX_TYPES(columns_map.scalar_type(), "reduce_sparse_csr_dim0_cpu_indices",
```

Comment:
`columns_map` dtype is always `kLong` (see `unique_cpu_template` in aten/src/ATen/native/Unique.cpp, for instance). So, please eliminate `AT_DISPATCH_INDEX_TYPES` usage as unnecessary. For that, just remove this and the following line, and replace `index_t` with `int64_t`.